### ISSUE TITLE
[BUG] Fails to Identify Global Value in HEREDOC

### ISSUE THREAD

**Issue by ndunn990** [mineiros-io/terramate#803]
**Describe the bug**
I  believe Terramate fails to resolve global values correctly when it's part of a `HEREDOC`. Fortunately, it's not difficult to work around in the meantime.

**To Reproduce**

Just as a quick example, I have a stack configuration that includes a global value `env`...
```
globals {
  env = "my_environment"
}
```

Create a `main.tm.hcl` file for a stack.
```
generate_hcl "_main.tf" {
  content {
    module "webapp_elastic_search" {
      source                   = "${terramate.stack.path.to_root}/modules/aws-elasticsearch"
      domain_name              = "${global.env}-web"
      enabled                  = true
      cloudwatch_log_enabled   = false
      create_service_link_role = false
      vpc_id                   = data.terraform_remote_state.vpc.outputs.primary_vpc_id
      vpc_options_subnet_ids   = [data.terraform_remote_state.vpc.outputs.primary_vpc_public_subnets[0]]
      vpc_allowed_cidr_blocks  = [data.terraform_remote_state.vpc.outputs.primary_vpc_cidr]
      access_policies          = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "es:*",
      "Principal": {
        "AWS": "*"
    },
      "Effect": "Allow",
      "Resource": "arn:aws:es:us-west-2:${data.aws_caller_identity.current.account_id}:domain/${global.env}-web/*"
    }
  ]
}
POLICY
    }
  }
}
```
When the file is generated, the `HEREDOC` looks like this with `${global.env}` still there.

```
// TERRAMATE: GENERATED AUTOMATICALLY DO NOT EDIT

module "webapp_elastic_search" {
  access_policies          = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": "es:*",
      "Principal": {
        "AWS": "*"
    },
      "Effect": "Allow",
      "Resource": "arn:aws:es:us-west-2:${data.aws_caller_identity.current.account_id}:domain/${global.env}-web/*"
    }
  ]
}
POLICY
  cloudwatch_log_enabled   = false
  create_service_link_role = false
  domain_name              = "my_environment-web"
  enabled                  = true
  source                   = "../../../modules/aws-elasticsearch"
  vpc_allowed_cidr_blocks = [
    data.terraform_remote_state.vpc.outputs.primary_vpc_cidr,
  ]
  vpc_id = data.terraform_remote_state.vpc.outputs.primary_vpc_id
  vpc_options_subnet_ids = [
    data.terraform_remote_state.vpc.outputs.primary_vpc_public_subnets[0],
  ]
}

```
**Expected behavior**
I expected the `${global.env}` text to instead be `my_environment` inside of the `HEREDOC`.  Unless there's a feature I'm not aware of or misunderstand.

**Log Output**
Terramate generation is successful, so no helpful logs.

**Environment (please complete the following information):**
 - OS: Linux
 - OS Version: Ubuntu 21.04
 - Git Version: 2.34.1
 - Terramate Version: 0.2.8
 
**Additional context**
None

**Comment by i4ki** [comment]
thanks for reporting this @ndunn990 

It's indeed a bug. We will provide a fix ASAP.

**Comment by i4ki** [comment]
The partial evaluation inside `HEREDOC` was actually never implemented.
Fixed on #805 but as we have fuzzing for this feature, we will need at least 2 days before we can be sure it covers all valid cases of interpolation inside HEREDOCs. It's going to be included on next release, for sure (`v0.2.9`).

**Comment by ndunn990** [comment]
Excellent. Thank you all for your hard work. Terramate is a fantastic tool.
